integration-tests/java/src/integration-test/java/com/stellarstation/api/test/satellite/SatelliteStreamerTest.java

```diff
@@ -124,19 +115,83 @@ public void onCompleted() {}
             .setEnableEvents(true)
             .build());
 
-    ListenableFuture<Boolean> timeoutFuture =
-        Futures.withTimeout(
-            commandTestFuture, 30, TimeUnit.SECONDS, Executors.newSingleThreadScheduledExecutor());
+    TimeUnit.SECONDS.sleep(20);
+    requestObserver.onCompleted();
 
-    // Check safe mode state is changed..
+    ListenableFuture<Boolean> timeoutFuture = createTimeoutFuture(testCompletionFuture, 3);
     assertThat(timeoutFuture.get()).isTrue();
 
-    // Check antenna states are valid.
-    assertThat(azimuthCommand).containsOnly(1.0);
-    assertThat(azimuthMeasured).containsOnly(1.02);
-    assertThat(elevationCommand).containsOnly(20.0);
-    assertThat(elevationMeasured).containsOnly(19.5);
+    assertThat(responseObserver.getSafeModeStates()).containsExactlyInAnyOrder(0, 1);
+  }
+
+  @Test
+  void events() throws Exception {
+    final SettableFuture<Boolean> testCompletionFuture = SettableFuture.create();
+
+    class EventTestStreamObserver implements StreamObserver<SatelliteStreamResponse> {
+      private List<AntennaState> antennaStates = new ArrayList();
+
+      @Override
+      public void onNext(SatelliteStreamResponse response) {
+        if (response.hasStreamEvent()) {
```

Comment:
We could consider checking the size of antennaStates here and only store the result of the first onNext?

As mentioned elsewhere, also calling the request observers onCompleted might be cleaner than the artificial delay.